### PR TITLE
Add a mapping timeout disruption test

### PR DESCRIPTION
**Pull request by andreidan**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java [hunk 1]
```diff
@@ -48,7 +56,7 @@
  */
 @TestLogging("_root:DEBUG,org.elasticsearch.cluster.service:TRACE")
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
-@SQLTransportIntegrationTest.Slow
+//@SQLTransportIntegrationTest.Slow
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java]: Either remove or keep?

File: sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java [hunk 2]
```diff
@@ -236,4 +244,51 @@ public void testVerifyApiBlocksDuringPartition() throws Exception {
         // the test to fail due to unfreed resources
         ensureStableCluster(2, nonIsolatedNode);
     }
+
+    @Test
+    @Repeat(iterations = 30)
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java]: Do we really want to keep the repeat?

File: sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java [hunk 3]
```diff
@@ -236,4 +244,51 @@ public void testVerifyApiBlocksDuringPartition() throws Exception {
...
+    public void testMappingTimeout() throws Exception {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java]: Maybe we can extend the method a bit to clarify the scenario we're testing.

File: sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java (hunk 2 above)
Comment (andreidan) [reply sql/src/test/java/io/crate/integrationtests/disruption/discovery/MasterDisruptionIT.java]: No, it's just so we repeat the tests in the CI before we merge. Sorry should've mentioned the 2nd commit is a temporary one that will be dropped before merge.

**Pull request merged**